iothub_client/src/uamqp_messaging.c

```diff
@@ -430,46 +434,102 @@ int message_create_uamqp_encoding_from_iothub_message(IOTHUB_MESSAGE_HANDLE mess
 
 static int readPropertiesFromuAMQPMessage(IOTHUB_MESSAGE_HANDLE iothub_message_handle, MESSAGE_HANDLE uamqp_message)
 {
-    int return_value;
+    int result;
     PROPERTIES_HANDLE uamqp_message_properties;
     AMQP_VALUE uamqp_message_property;
-    const char* uamqp_message_property_value;
     int api_call_result;
 
     // Codes_SRS_UAMQP_MESSAGING_09_008: [The uAMQP message properties shall be retrieved using message_get_properties().]
     if ((api_call_result = message_get_properties(uamqp_message, &uamqp_message_properties)) != 0)
     {
         // Codes_SRS_UAMQP_MESSAGING_09_009: [If message_get_properties() fails, message_create_IoTHubMessage_from_uamqp_message shall fail and return immediately.]
         LogError("Failed to get property properties map from uAMQP message (error code %d).", api_call_result);
-        return_value = __FAILURE__;
+        result = __FAILURE__;
     }
     else
     {
-        return_value = 0; // Properties 'message-id' and 'correlation-id' are optional according to the AMQP 1.0 spec.
+        result = 0; // Properties 'message-id' and 'correlation-id' are optional according to the AMQP 1.0 spec.
 
         // Codes_SRS_UAMQP_MESSAGING_09_010: [The message-id property shall be read from the uAMQP message by calling properties_get_message_id.]
         if ((api_call_result = properties_get_message_id(uamqp_message_properties, &uamqp_message_property)) != 0)
         {
             // Codes_SRS_UAMQP_MESSAGING_09_011: [If properties_get_message_id fails, message_create_IoTHubMessage_from_uamqp_message() shall fail and return immediately.]
             LogInfo("Failed to get value of uAMQP message 'message-id' property (%d). No failure, since it is optional.", api_call_result);
```

Comment:
>LogInfo("Failed to get value of uAMQP message 'message-id' property (%d). No failure, since it is optional.", api_call_result); [](start = 12, length = 127)

If this is actually optional do we need to logInfo?

Reply:
Yes, we do.
When we try to get the value, the function can succeed and return NULL (because it's optional) and then in that case we are ok, there is no error to log.
But the function can also fail (return non-zero), and that case is unexpected, so we need to treat as a failure.

Reply:
I take that back. Sorry for the misunderstanding. Fixed now.